app/api/lectures.js

```diff
@@ -0,0 +1,276 @@
+            }
+        }
+        catch (e) {
+            res.status(400).send({error: "Unable to create association between lecture & this course' sections"})
+            return
+        }
+        res.status(201).json(lecture)   // all good, return lecture object
+    }
+    else {      // if user is not a teacher
+        res.status(403).send()
+    }
+})
+
+router.put('/:lecture_id', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub)
+    const lectureId = req.params.lecture_id
+    const courseId = req.params.course_id
+    const enrollment = await getEnrollmentFromCourse(user.id, courseId) || await getEnrollmentFromSectionInCourse(user.id, courseId)
+    const lecture = await getLecture(lectureId)
+    
+    if (lecture == null) {  // if passed in lectureId does not exist
+        res.status(404).send({error: "Lecture of this id does not exist"})
+        return
+    }
+    else if (enrollment == null) {   // if user is not enrolled in this course
+        res.status(403).send()
+    }
+    else if (enrollment.role == 'teacher') {
+        try {
+            await db.Lecture.update(
+                req.body,     // provided body from request
+                { where: { id: lectureId } }
+            )
+        }
+        catch (e) {
+            res.status(400).send({error: "Unable to update lecture object"})
+            return
+        }
+        res.status(200).send()   // all good, return updated lecture object
+    }
+    else {      // if user is not a teacher
+        res.status(403).send()
+    }
+})
+
+router.get('/:lecture_id', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub)
+    const lectureId = req.params.lecture_id
+    const courseId = req.params.course_id
+    const enrollment = await getEnrollmentFromCourse(user.id, courseId) || await getEnrollmentFromSectionInCourse(user.id, courseId)
+    const lecture = await getLecture(lectureId)
+    var full_response = {}  // will hold response with lecture info and related questions
+
+    if (enrollment == null) {   // if user is not enrolled in this course
+        res.status(403).send()
+        return
+    }
+    else if (lecture == null) {      // if passed in lectureId does not exist
+        res.status(404).send({error: "Lecture of this id does not exist"})
+        return
+    }
+    else if (enrollment.role == 'teacher') {     // if teacher, send lecture info & all related questions
+        try {
+            full_response['lecture'] = lecture  // full_response will hold wanted lecture along with its related questions in the end
+            const questions_in_lec = await db.sequelize.query(  // raw sql query to get all questions in this lecture using `QuestionInLecture`
+                `SELECT q.* FROM Questions q INNER JOIN QuestionInLectures ql ON q.id = ql.questionId INNER JOIN Lectures l ON ql.lectureId = l.id WHERE l.id = ${lectureId}`,
```

Comment:
We have to make sure this is protected from SQL Injection. I imagine sequelize comes with that out of the box but raw queries make me nervous